File: app/src/main/java/com/anymore/client/di/AppModule.java
package com.anymore.client.di;

import android.app.Application;
import dagger.Module;
import dagger.Provides;

/**
 * Created by liuyuanmao on 2019/7/16.
 */
@Module(includes = ActivityBindingModule.class)
public class AppModule {
    private Application mApplication;

    public AppModule(Application application) {
        this.mApplication = application;
    }

    @Provides
    public Application provideApplication(){
        return mApplication;
    }
}

File: app/src/main/java/com/anymore/client/di/module/AppModule.kt
package com.anymore.client.di.module

import android.app.Application
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by liuyuanmao on 2019/7/23.
 */
@Module(includes = [ActivityBindingModule::class,FragmentBindingModule::class])
class AppModule (private val application: Application){

    @Singleton
    @Provides
    fun provideApplication()=application
}